src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/controller/bfd/BfdSessionFsm.java

```diff
@@ -364,335 +443,220 @@ private void reportActionFailure(BfdSessionAction.ActionResult result) {
                     // extra parameters
-                    PersistenceManager.class, Endpoint.class, Integer.class);
+                    PersistenceManager.class, SwitchOnlineStatusMonitor.class, EndpointStatusMonitor.class,
+                    IBfdSessionCarrier.class, Endpoint.class, Integer.class, BfdSessionData.class);
 
-            // INIT
-            builder.transition()
-                    .from(State.INIT).to(State.INIT_CHOICE).on(Event.HISTORY)
-                    .callMethod("consumeHistory");
+            final String sessionRemoveAction = "sessionRemoveAction";
+            final String emitBfdFailAction = "emitBfdFailAction";
 
-            // INIT_CHOICE
-            builder.transition()
-                    .from(State.INIT_CHOICE).to(State.IDLE).on(Event._INIT_CHOICE_CLEAN);
-            builder.transition()
-                    .from(State.INIT_CHOICE).to(State.INIT_REMOVE).on(Event._INIT_CHOICE_DIRTY);
-            builder.onEntry(State.INIT_CHOICE)
-                    .callMethod("handleInitChoice");
-
-            // IDLE
-            builder.transition()
-                    .from(State.IDLE).to(State.INIT_SETUP).on(Event.ENABLE_UPDATE)
-                    .callMethod(saveIslReferenceMethod);
-            builder.transition()
-                    .from(State.IDLE).to(State.UNOPERATIONAL).on(Event.OFFLINE);
-            builder.onEntry(State.IDLE)
-                    .callMethod("idleEnter");
-
-            // UNOPERATIONAL
-            builder.transition()
-                    .from(State.UNOPERATIONAL).to(State.IDLE).on(Event.ONLINE);
-            builder.transition()
-                    .from(State.UNOPERATIONAL).to(State.PENDING).on(Event.ENABLE_UPDATE)
-                    .callMethod(savePropertiesMethod);
-
-            // PENDING
-            builder.transition()
-                    .from(State.PENDING).to(State.UNOPERATIONAL).on(Event.DISABLE);
-            builder.transition()
-                    .from(State.PENDING).to(State.INIT_SETUP).on(Event.ONLINE);
-            builder.onEntry(State.PENDING)
-                    .callMethod(saveIslReferenceMethod);
-
-            // INIT_SETUP
-            builder.transition()
-                    .from(State.INIT_SETUP).to(State.IDLE).on(Event.FAIL);
-            builder.transition()
-                    .from(State.INIT_SETUP).to(State.DO_SETUP).on(Event.NEXT);
-            builder.onEntry(State.INIT_SETUP)
-                    .callMethod("doAllocateResources");
-
-            // DO_SETUP
-            builder.transition()
-                    .from(State.DO_SETUP).to(State.ACTIVE).on(Event.ACTION_SUCCESS)
-                    .callMethod("reportSetupSuccess");
-            builder.transition()
-                    .from(State.DO_SETUP).to(State.INIT_REMOVE).on(Event.DISABLE);
+            // ENTER
+            builder.onEntry(State.ENTER)
+                    .callMethod("enterEnterAction");
```

Comment:
Maybe we can rename `Enter` FSM state to `Init` or something else to avoid `enterEnter` in method name?

Reply:
... I would not do it. Besides I don't see issues in `enterEnter` name.